src/lib/account_update.ts

```diff
@@ -1065,34 +1025,49 @@ class AccountUpdate implements Types.AccountUpdate {
     }
   }
 
-  toPublicInput(): ZkappPublicInput {
+  toPublicInput({
+    accountUpdates,
+  }: {
+    accountUpdates: AccountUpdate[];
+  }): ZkappPublicInput {
     let accountUpdate = this.hash();
-    let calls = CallForest.hashChildren(this);
+
+    // collect this update's descendants
+    let descendants: AccountUpdate[] = [];
+    let callDepth = this.body.callDepth;
+    let i = accountUpdates.findIndex((a) => a.id === this.id);
+    assert(i !== -1, 'Account update not found in transaction');
```

Comment:
Do you think it would be helpful to print the ID of the AU that wasn't found? Not sure if it would actually be helpful for debugging :shrug:

Reply:
I'm leaving this out for now since it's definitely not an error I expect will be encountered - we can improve the message if someone hits this